kie-remote/kie-remote-client/src/test/java/org/kie/services/client/builder/http/HeaderTest.java

```diff
@@ -0,0 +1,216 @@
+import org.simpleframework.common.buffer.Allocator;
+import org.simpleframework.common.buffer.FileAllocator;
+import org.simpleframework.http.Request;
+import org.simpleframework.http.Response;
+import org.simpleframework.http.core.Container;
+import org.simpleframework.http.core.ContainerTransportProcessor;
+import org.simpleframework.transport.SocketProcessor;
+import org.simpleframework.transport.TransportProcessor;
+import org.simpleframework.transport.TransportSocketProcessor;
+import org.simpleframework.transport.connect.Connection;
+import org.simpleframework.transport.connect.SocketConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ */
+public class HeaderTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(RedirectTest.class);
+
+    private static final String DEFAULT_ENPOINT = "/rest/execute";
+
+    private static final String TEST_HEADER_NAME = UUID.randomUUID().toString();
+    private static final String ANOTHER_TEST_HEADER_NAME = UUID.randomUUID().toString();
+    private static final String NOT_SENT_HEADER_NAME = UUID.randomUUID().toString();
+
+    private static final Random random = new Random();
+
+    @Test
+    public void testSetHeader() throws Exception {
+        int port = AvailablePortFinder.getNextAvailable(1025);
+        HttpHeaderServer server = new HttpHeaderServer(port);
+
+        server.start();
+
+        // @formatter:off
+        RuntimeEngine runtimeEngine =  RemoteRuntimeEngineFactory.newRestBuilder()
+            .addPassword("user").addUserName("pass")
+            .addDeploymentId("deploymentId")
+            .addUrl(new URL("http://localhost:" + port + "/" ))
+            .build();
+        // @formatter:on
+
+        // make sure our server works..
+        ProcessInstance procInst = runtimeEngine.getKieSession().startProcess("test");
+        assertNotNull( "Test server is not working..", procInst );
+        assertTrue( "No header, so no event types should be in the returned process instance!", procInst.getEventTypes() == null || procInst.getEventTypes().length == 0 );
+
+        String headerValueOne = String.valueOf(random.nextLong());
+        String headerValueTwo = String.valueOf(random.nextLong());
+
+        // @formater:off
+        runtimeEngine =  RemoteRuntimeEngineFactory.newRestBuilder()
+            .addPassword("user").addUserName("pass")
+            .addDeploymentId("deploymentId")
+            .addUrl(new URL("http://localhost:" + port + "/" ))
+            .addHeader(NOT_SENT_HEADER_NAME, String.valueOf(random.nextLong()))
+            .clearHeaderFields()
+            .addHeader(TEST_HEADER_NAME, headerValueOne)
+            .addHeader(ANOTHER_TEST_HEADER_NAME, headerValueTwo)
+            .build();
+        // @formatter:on
+
+        procInst = runtimeEngine.getKieSession().startProcess("test");
+        assertNotNull( "Test server is not working..", procInst );
```

Comment:
I would also consider having this assertion (somehow) in the ```@Before``` method - since when this fails, there's no point in executing the test.